File: src/eu/rafig/covid/core/game/entities/placeable/Organism.java
package eu.rafig.covid.core.game.entities.placeable;

public abstract class Organism {
    private double constant = 0;
    private String name = "";
    private Harmness harmness = Harmness.NEUTRAL;

    public Organism(double constant, Harmness harmness) {
        this.constant = constant;
        this.harmness = harmness;
    }

    public Organism(double constant, String name) {
        this.constant = constant;
        this.name = name;
    }

    public Organism(double constant) {
        this.constant = constant;
    }

    public double getConstant() {
        return constant;
    }

    public String getName() {
        return name;
    }

    public Harmness getHarmness() {
        return harmness;
    }
}

File: Assets/src/C#/entities/placeable/Organism.cs
using System.Collections;
using System.Collections.Generic;
using eu.parada.enums;

namespace eu.parada.entities.placeable {
    public abstract class Organism {
        public double constant { get; private set; }
        public string name { get; private set; }
        public HarmnesType harmness { get; private set; }

        public Organism(double constant, HarmnesType harmness) {
            this.constant = constant;
            this.harmness = harmness;
        }

        public Organism(double constant, string name) {
            this.constant = constant;
            this.name = name;
        }

        public Organism(double constant) {
            this.constant = constant;
        }
    }
}
